strawberry/channels/handlers/http_handler.py

```diff
@@ -185,16 +202,25 @@ def create_response(
     async def handle(self, body: bytes) -> None:
         request = ChannelsRequest(consumer=self, body=body)
         try:
-            response: ChannelsResponse = await self.run(request)
+            response = await self.run(request)
 
             if b"Content-Type" not in response.headers:
                 response.headers[b"Content-Type"] = response.content_type.encode()
 
-            await self.send_response(
-                response.status,
-                response.content,
-                headers=response.headers,
-            )
+            if isinstance(response, MultipartChannelsResponse):
+                response.headers[b"Transfer-Encoding"] = b"chunked"
+                await self.send_headers(headers=response.headers)
+
+                async for chunk in response.stream():
+                    # TODO: we should change more body
+                    await self.send_body(chunk.encode("utf-8"), more_body=True)
+
+            else:
```

Comment:
Maybe do `elif isinstance(response, ChannelsResponse)` with `else: assert_never(response)` to make sure that we are handling every possibility in typing and runtime